homer0/projext-plugin-webpack-aurelia
# v1.0.1: fix(project): implement a custom HTML loader

### What does this PR do?

[`projext-plugin-webpack`](https://yarnpkg.com/en/package/projext-plugin-webpack) version `8` updated [`raw-loader`](https://yarnpkg.com/en/package/raw-loader) to its latest version, which instead of exporting file contents with `module.exports`, started exporting them with ES modules syntax (`export default`). The problem with that is that the [`aurelia-loader`](https://yarnpkg.com/en/package/aurelia-loader) uses `require` to load the HTML files, and the whole thing was breaking because it doesn't check for a `default` property.

The solution was to add a custom HTML loader before the `aurelia-loader` that simply checks if the file contents matches `/^\s*export\s*default\s*["|']\s*<template\s*/i` and in case it does, replace it with `module.exports`.

And I also did two small things:

- I updated all the dependencies to their latest versions.
- I fixed the PR template.

### How should it be tested manually?

First, to verify the error, try to run your app with the latest version of [`projext-plugin-webpack`](https://yarnpkg.com/en/package/projext-plugin-webpack) and the latest published version of this plugin. It should throw an error on runtime that a template contents are invalid.

Now, to try the fix, just install this branch; everything should go back to normal.

And of course....

```bash
yarn test
# or
npm test
```
